### PR TITLE
[FTAMS102020] Rich Alossi

### PR DESCRIPTION
**Pull request by alossix**

### CODE REVIEW

File: js/memory.js [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+  }
+  shuffleCards(cardsObj) {
+    if (!cardsObj) {
+      return undefined;
+    }
+    for (let i = 0; i < cardsObj.length; i++) {
+      let rand = Math.floor(Math.random() * cardsObj.length);
+      cardsObj.push(cardsObj[rand]);
+      cardsObj.splice(rand, 1);
+    }
```
Comment (wax-bear) [inline js/memory.js:16]: I see what you do here, and it makes sense.
Would there be an alternative way that ensures that every card is shuffled, perhaps by making use of the index iterated through?

File: js/memory.js [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
   }
-  shuffleCards() {}
   checkIfPair(card1, card2) {}
   isFinished() {}
-}
\ No newline at end of file
+}
+
+let cardsArrTemp = cardsArr;
+for (let i = 0; i < cardsArrTemp.length; i++) {
```
Comment (wax-bear) [inline js/memory.js:23]: If this is double code, I would remove it.

File: js/memory.js [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+  let rand = Math.floor(Math.random() * cardsArrTemp.length);
+  cardsArrTemp.push(cardsArr[rand]);
+  cardsArr.splice(rand, 1);
+}
```
Comment (wax-bear) [inline js/memory.js:27]: Great stuff Rich! 
If you have any questions please reach out to me.

File: js/memory.js (hunk 2 above)
Comment (alossix) [reply js/memory.js:23]: Thanks, yes, double-code now removed :)

**Pull request closed without merging**